test/unit/spec/v1/Registration_spec.js

```diff
@@ -1037,4 +1037,126 @@ Expect.describe('Registration', function() {
+            preSubmitSpy(postData, onSuccess, onFailure);
+            onSuccess(postData);
+          },
+          postSubmit: jasmine.createSpy('postSubmitSpy'),
+        },
+      };
+
+      return setup(setting).then(function(test) {
+        Util.resetAjaxRequests();
+        test.form.setUserName('test@example.com');
+        test.form.setPassword('Abcd1234');
+        test.form.setFirstname('firstName');
+        test.form.setLastname('');
+        test.form.submit();
+        const model = test.router.controller.model;
+
+        spyOn(Backbone.Model.prototype, 'save').and.returnValue($.Deferred().resolve());
+        model.save();
+        Util.callAllTimeouts();
+        expect(test.router.controller.model.get('userName')).toBe('test@example.com');
+        expect(test.router.controller.model.get('firstName')).toBe('firstName');
+        expect(test.router.controller.model.has('lastName')).toBe(false);
+        expect(setting.registration.postSubmit).toHaveBeenCalled();
+      });
+    });
+    itp('undefined value', function() {
+      const preSubmitSpy = jasmine.createSpy('preSubmitSpy');
+      const setting = {
+        registration: {
+          preSubmit: function(postData, onSuccess, onFailure) {
+            preSubmitSpy(postData, onSuccess, onFailure);
+            onSuccess(postData);
+          },
+          postSubmit: jasmine.createSpy('postSubmitSpy'),
+        },
+      };
+
+      return setup(setting).then(function(test) {
+        Util.resetAjaxRequests();
+        test.form.setUserName('test@example.com');
+        test.form.setPassword('Abcd1234');
+        test.form.setFirstname('firstName');
+        test.form.setLastname(undefined);
+        test.form.submit();
+        const model = test.router.controller.model;
+
+        spyOn(Backbone.Model.prototype, 'save').and.returnValue($.Deferred().resolve());
+        model.save();
+        Util.callAllTimeouts();
+        expect(test.router.controller.model.get('userName')).toBe('test@example.com');
+        expect(test.router.controller.model.get('firstName')).toBe('firstName');
+        expect(test.router.controller.model.has('lastName')).toBe(false);
+        expect(setting.registration.postSubmit).toHaveBeenCalled();
+      });
+    });
+    itp('no sanitize', function() {
+      const preSubmitSpy = jasmine.createSpy('preSubmitSpy');
+      const setting = {
+        registration: {
+          preSubmit: function(postData, onSuccess, onFailure) {
+            preSubmitSpy(postData, onSuccess, onFailure);
+            onSuccess(postData);
+          },
+          postSubmit: jasmine.createSpy('postSubmitSpy'),
+        },
+      };
+
+      return setup(setting).then(function(test) {
+        Util.resetAjaxRequests();
+        test.form.setUserName('test@example.com');
+        test.form.setPassword('Abcd1234');
+        test.form.setFirstname('firstName');
+        test.form.setLastname('lastName');
```

Comment:
Curious, would it be possible here to add a boolean field to the form and verify the logic works on it too? Just asking since that's where the bug originated from

Reply:
Discussed offline, resolving !